tests/test_quantity.py

```diff
@@ -766,6 +766,35 @@ def test_temp_addition_69():
     assert t.type == "Temperature"
 
 
+def test_quantity_map_70():
+    quantity_map_from_settings_API = {
+        "Mass": 1,
+        "Velocity": 2.5,
+        "Current": 3,
+        "Light": 2,
+        "Epsilon Flux Coefficient": 2,
+    }
+
+    api_test = q.Quantity(10.5, quantity_map=quantity_map_from_settings_API)
+    assert api_test.value == 10.5
+    assert api_test.unit == "kg m^2.5 s^-2.5 A^3 cd^2 kg^2 m^-4 s^-4"
```

Comment:
Ideally, terms with the same units should be collected together. For instances, metres and seconds appear twice each.

Reply:
by collecting terms with the same units together will only reduce the unit and need to have another loop and power sum method for each base and it will make this operation too much expensive.

Currently each term in `final_unit` (`"kg m^2.5 s^-2.5 A^3 cd^2 kg^2 m^-4 s^-4"`) returned by `get_unit_from_map` method is handled one by one through parser.

So there is no need to collect terms with same units together.

Reply:
> Ideally, terms with the same units should be collected together. For instances, metres and seconds appear twice each.

One another point is, this returned unit is not reduced to SI fundamentals units and hence we will have to consider both SI bases and derived units bases like `N (newton), T (Tesla), H (Henry), Wb (weber)` etc.

Reply:
@seanpearsonuk  Done. Thank you.